app/controllers/users/sessions_controller.rb

```diff
@@ -1,28 +1,64 @@
 class Users::SessionsController < Devise::SessionsController
   before_action :configure_sign_in_params, only: [:create]
-
-  # GET /resource/sign_in
-  # def new
-  #   super
-  # end
+  skip_before_action :verify_authenticity_token, only: [:renew_jwt]
 
   # POST /resource/sign_in
   def create
     if params.dig(:user, :email).present? && (params[:user][:email].exclude? "@")
       params[:user][:email] << "@sanger.ac.uk"
     end
+
     super
+
+    set_session_cookie
+
+    jwt = prepare_jwt_cookie({"email": params[:user][:email], "groups": current_user.groups})
+    cookies[:aker_user_jwt] = JWT.encode jwt, secret_key, 'HS256'
   end
 
   # DELETE /resource/sign_out
-  # def destroy
-  #   super
-  # end
+  def destroy
+    super
+    cookies.delete :aker_user_jwt
+    session.destroy
+  end
+
+  def renew_jwt
+    # Check session is valid
+    # Renew JWT if so
+    # Otherwise, unauthorized error
+
+    if session[:email].present?
+      jwt = prepare_jwt_cookie({email: session[:email], groups: User.find_by(email: session[:email]).groups})
+      
+      coded_jwt = JWT.encode jwt, secret_key, 'HS256'
+      cookies[:aker_user_jwt] = coded_jwt
```

Comment:
From what I've read in the docs here:
http://api.rubyonrails.org/v5.1/classes/ActionDispatch/Cookies.html
we have to specify the expiring period for the cookie on creation, otherwise it will be created as a session cookie, so it will disappear after closing the browser. We can also provide some other config that could be interesting:
:secure -> true, so it will only be sent on https requests
:httponly -> true, so it won't be readable by javascript, which is fine because we are not using its value
:expires -> this one should be the same as the session in database, 1 month?